pig/test/org/partiql/pig/domain/TypeDomainParserIncludeFileTests.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package org.partiql.pig.domain
+
+import com.amazon.ionelement.api.IonTextLocation
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.partiql.pig.domain.model.TypeDomain
+import org.partiql.pig.domain.model.TypeUniverse
+import org.partiql.pig.domain.parser.ParserErrorContext
+import org.partiql.pig.domain.parser.SourceLocation
+import org.partiql.pig.domain.parser.parseMainTypeUniverse
+import org.partiql.pig.errors.PigError
+import org.partiql.pig.errors.PigException
+import java.nio.file.Paths
+import kotlin.test.assertTrue
+
+class TypeDomainParserIncludeFileTests {
+    private val mainDir = Paths.get("test-domains").toAbsolutePath()
+    private val rootA = mainDir.resolve("root_a").toAbsolutePath()
+    private val rootB = mainDir.resolve("root_b").toAbsolutePath()
+
+    @Test
+    fun `include happy case`() {
+        val universe = parseWithTestRoots("test-domains/main.ion")
+        val allDomains = universe.statements.filterIsInstance<TypeDomain>()
```

Comment:
This was brought up offline in a team discussion and in Abhilash's review. What happens in the case where an included domain has the same name as a previously defined domain?

For example, if `universe_b.ion` had defined `domain_a` again and not `domain_b`?

```
// defining `domain_a` (which has already been defined in `universe_a.ion`) rather than `domain_b`
(define domain_a (domain (product foo)))
```

Would this be an error or overridden? If it's an error, would this be thrown by the `IncludeResolver`? I think a test and/or a comment should be added detailing this behavior.

Reply:
Currently when I do the change (i.e. make `domain_b` -> `domain_a` in `universe_b.ion`), no error is given for a duplicated domain name.